packages/core/src/__tests__/shutdown-signal.test.ts

```diff
@@ -6,7 +6,13 @@ const fn = (abortSignal: AbortSignal) => {
     abortSignal.addEventListener('abort', () => {
       reject(new Error(`aborted after run`))
     })
-    new Promise((resolve1) => setTimeout(resolve1, 500)).then(() => resolve('ok'))
+    new Promise((resolve1) => setTimeout(resolve1, 500))
+      .then(() => {
+        resolve('ok')
+      })
+      .catch((error) => {
+        reject(error)
+      })
```

Comment:
Seems unnecessary to have a promise inside a promise here, I think it could be simplified as

```suggestion
    setTimeout(() => {
      resolve('ok')
    }, 500)
```